src/OpenTelemetry.Extensions.Enrichment/TraceEnrichmentProcessor.cs

```diff
@@ -0,0 +1,61 @@
+// <copyright file="TraceEnrichmentProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Extensions.ObjectPool;
+
+namespace OpenTelemetry.Extensions.Enrichment;
+
+internal sealed class TraceEnrichmentProcessor : BaseProcessor<Activity>
+{
+    private readonly List<BaseTraceEnricher> traceEnrichers = new();
```

Comment:
Sure, it can be done. However, we have the AddEnricher() method. In case we use array, it will look like:

```
public void AddEnricher(BaseTraceEnricher traceEnricher)
{
    Debug.Assert(traceEnricher != null, "trace enricher was null");

    Array.Resize(ref this.traceEnrichers, this.traceEnrichers.Length + 1);
    this.traceEnrichers[this.traceEnrichers.Length - 1] = traceEnricher!;
}
```

It is not a hot path anyway, but still, I am not sure how appropriate it is to declare an array but then resize it every time :)

Reply:
The point is that for DI you will have all enrichers by the time you hit the constructor so no resize is needed, for non DI yes there will be resize but that's at the startup time and one time cost. We want to ensure the hotpath doesn't need to pay the penalty

Reply:
Updated to array